.github/workflows/get-environment.yml

```diff
@@ -0,0 +1,321 @@
+                targetStability = getStability(branchName);
+
+                if (targetStability !== 'canary') {
+                  return false;
+                }
+
+                return true;
+              });
+
+              core.setOutput('target_stability', targetStability);
+            }
+
+            core.setOutput('is_targeting_feature_branch', isTargetingFeatureBranch);
+
+      - name: Get version from ${{ inputs.version_file }}
+        id: get_version
+        run: |
+          if [[ "${{ inputs.version_file }}" == */.version ]]; then
+            . .version
+            . ${{ inputs.version_file }}
+            VERSION="$MAJOR.$MINOR"
+          elif [[ "${{ inputs.version_file }}" == */insertBaseConf.sql ]]; then
+            VERSION=$(grep version ${{ inputs.version_file }} | cut -d "'" -f 4 | awk 'NR==2')
+          elif [[ "${{ inputs.version_file }}" == */conf.php ]]; then
+            VERSION=$(cat ${{ inputs.version_file }}| grep mod_release | cut -d'=' -f2 | cut -d'"' -f2 | cut -d"'" -f2)
+          elif [[ "${{ inputs.version_file }}" == */configs.xml ]]; then
+            VERSION=$(sed -n 's|\s*<version>\(.*\)</version>|\1|p' ${{ inputs.version_file }} 2>/dev/null)
+          elif [[ "${{ inputs.version_file }}" == */pom.xml ]]; then
+            VERSION=$(grep '<revision>.*</revision>' ${{ inputs.version_file }} | cut -d '>' -f 2 | cut -d - -f 1)
+          elif [[ "${{ inputs.version_file }}" == */*.spectemplate ]]; then
+            VERSION=$(grep 'Version: ' ${{ inputs.version_file }} | rev | cut -d' ' -f1 | rev)
+          elif [[ "${{ inputs.version_file }}" == */.env ]]; then
+            VERSION=$(sed -e '/\[[2-9][0-9].[0-9][0-9].[0-9]+\]/p' ${{ inputs.version_file }} | cut -d '=' -f 2)
+          else
+            echo "Unable to parse version file ${{ inputs.version_file }}"
+            exit 1
+          fi
+
+          if egrep '^[2-9][0-9]\.[0-9][0-9]\.[0-9]+' <<<"$VERSION" >/dev/null 2>&1 ; then
+            n=${VERSION//[!0-9]/ }
+            a=(${n//\./ })
+            echo "major_version=${a[0]}.${a[1]}" >> $GITHUB_OUTPUT
+            MAJOR=${a[0]}.${a[1]}
+            echo "minor_version=${a[2]}" >> $GITHUB_OUTPUT
+          else
+            echo "Cannot parse version number from ${{ inputs.version_file }}"
+            exit 1
+          fi
+        shell: bash
+
+      - name: "Get release: 1 for testing / stable, <date>.<commit_sha> for others"
+        id: get_release
+        run: |
+          if [[ "${{ steps.get_stability.outputs.stability }}" == "testing" || "${{ steps.get_stability.outputs.stability }}" == "stable" ]]; then
+            RELEASE="1"
+          else
+            RELEASE="$(date +%s).$(echo ${{ github.sha }} | cut -c -7)"
+          fi
+
+          echo "release=$RELEASE" >> $GITHUB_OUTPUT
+        shell: bash
+
+      - name: "Get release type: hotfix, release or not defined if not a release"
+        id: get_release_type
```

Comment:
is it really "safe" to not define the release type in every case here?

Reply:
Yes if value is properly surrounded by quotes in bash scripts